pilot/pkg/model/service.go

```diff
@@ -623,6 +623,11 @@ func (ep *IstioEndpoint) FirstAddressOrNil() string {
 	return ep.Addresses[0]
 }
 
+// Key returns a function suitable for usage to distinguish this IstioEndpoint from another
+func (ep *IstioEndpoint) Key() string {
+	return ep.FirstAddressOrNil() + "/" + ep.WorkloadName
```

Comment:
Can this still overlap in a strange case: You have a ServiceEntry with inline WorkloadEntry with the same name and IP as a WorkloadEntry. 

It could happen if you're moving from inline to selectors.